esp/esdllib/esdl_script.cpp

```diff
@@ -771,111 +829,129 @@ class CEsdlTransformOperationHttpPostXml : public CEsdlTransformOperationBase
         targetContext->addElementToLocation("headers");
         ForEachItemIn(i, m_headers)
             m_headers.item(i).processHeader(scriptContext, targetContext, sourceContext, headers);
+        if (!headers->hasProp("Accept"))
+            headers->setProp("Accept", "text/html, application/xml");
     }
 
-    void buildRequest(IEsdlScriptContext * scriptContext, IXpathContext * targetContext, IXpathContext * sourceContext, const char *url, IProperties *headers)
+    void buildRequest(OperationStateHttpPostXml &preparedState, IEsdlScriptContext * scriptContext, IXpathContext * targetContext, IXpathContext * sourceContext)
     {
         CXpathContextLocation location(targetContext);
         targetContext->addElementToLocation("request");
-        targetContext->ensureSetValue("@url", url, true);
-        buildHeaders(scriptContext, targetContext, sourceContext, headers);
+        targetContext->ensureSetValue("@url", preparedState.url, true);
+        buildHeaders(scriptContext, targetContext, sourceContext, preparedState.headers);
         if (m_content)
             m_content->process(scriptContext, targetContext, sourceContext);
+        VStringBuffer xpath("/esdl_script_context/%s/%s/request/content/*[1]", m_section.str(), m_name.str());
+        sourceContext->toXml(xpath, preparedState.content);
     }
 
-    void postRequest(IEsdlScriptContext * scriptContext, IXpathContext * targetContext, IXpathContext * sourceContext, const char *url, IProperties *headers)
+    virtual IInterface *prepare(IEsdlScriptContext * scriptContext, IXpathContext * targetContext, IXpathContext * sourceContext) override
     {
-        VStringBuffer xpath("/esdl_script_context/%s/%s/request/content/*[1]", m_section.str(), m_name.str());
-
-        StringBuffer content;
-        sourceContext->toXml(xpath, content);
-        if (!content)
-            return;
+        Owned<OperationStateHttpPostXml> preparedState = new OperationStateHttpPostXml;
         CXpathContextLocation location(targetContext);
-        targetContext->addElementToLocation("response");
+        VStringBuffer xpath("/esdl_script_context/%s/%s", m_section.str(), m_name.str());
+        targetContext->ensureLocation(xpath, true);
+        if (m_url)
+            sourceContext->evaluateAsString(m_url, preparedState->url);
+        if (m_testDelay)
```

Comment:
Would it make sense to make this conditional on being in a test mode? The script context knows to use printf output, which is currently a good indicator of being used for testing.

Reply:
Do we have a test mode, or are you saying to add one? I think there could be danger in leaving an attribute like this in code that goes to production. What would be the plumbing and work needed for a test mode? Is that a compile flag or something else?

Reply:
We have an implied test mode. By default, trace output is routed through jlog functions but a script context property is set by the unit tests causing the output to be generated by printf. This property, which cannot be changed through scripting or configuration, implies that we are in a test mode when set.

My concern with the `testDelay` script property is that it can be used in production environments. If we're OK with that, it's probably misnamed. If we don't want it used outside of tests, we already have a reliable way to limit it.

Reply:
done.